### PR TITLE
Added opentracing spans to calls to tokeninfo and tokenintrospection endpoints

### PR DESCRIPTION
**Pull request by arjunrn**
Using features implemented in PR #890 added opentracing into calls to the tokeninfo and tokenintrospection backends.

Signed-off-by: Arjun Naik <arjun.rn@gmail.com>

### CODE REVIEW

File: filters/auth/authclient.go [hunk 1]
```diff
@@ -9,12 +9,15 @@ import (
...
+	"strings"
 	"sync"
 	"time"
 )
 
 const (
-	webhookSpanName = "webhook"
+	webhookSpanName        = "webhook"
+	tokenInfoSpanName      = "tokeninfo"
+	tokenIntrospectionSpan = "tokenIntrospection"
```
Comment (szuecs) [inline filters/auth/authclient.go]: camelcase span -> better all lowercase

**Comment by szuecs** [comment]
Everything else gtm

File: filters/auth/authclient.go [hunk 2]
```diff
@@ -108,13 +111,18 @@ func createHTTPClient(timeout time.Duration, quit chan struct{}) (*http.Client,
 }
 
 // jsonGet does a get to the url with accessToken as the bearer token in the authorization header. Writes response body into doc.
-func jsonGet(url *url.URL, accessToken string, doc interface{}, client *http.Client) error {
+func jsonGet(url *url.URL, accessToken string, doc interface{}, client *http.Client, tracer opentracing.Tracer, parentSpan opentracing.Span, childSpanName string) error {
```
Comment (aryszka) [inline filters/auth/authclient.go]: could be broken into lines:

```
func jsonGet(
    url *url.URL,
    accessToken string,
    doc interface{},
    client *http.Client,
    tracer opentracing.Tracer,
    parentSpan opentracing.Span,
    childSpanName string,
) error {
```

File: filters/auth/authclient.go [hunk 3]
```diff
@@ -129,11 +137,24 @@ func jsonGet(url *url.URL, accessToken string, doc interface{}, client *http.Cli
...
+		span := tracer.StartSpan(childSpanName, opentracing.ChildOf(parentSpan.Context()))
+		tracer.Inject(span.Context(), opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(req.Header))
+		return span
+	}
+	return nil
+}
+
 // jsonPost does a form post to the url with auth in the body if auth was provided. Writes response body into doc.
-func jsonPost(u *url.URL, auth string, doc *tokenIntrospectionInfo, client *http.Client) error {
+func jsonPost(u *url.URL, auth string, doc *tokenIntrospectionInfo, client *http.Client, tracer opentracing.Tracer, parentSpan opentracing.Span, spanName string) error {
```
Comment (aryszka) [inline filters/auth/authclient.go]: could be broken into lines:

```
func jsonPost(
    u *url.URL,
    auth string,
    doc *tokenIntrospectionInfo,
    client *http.Client,
    tracer opentracing.Tracer,
    parentSpan opentracing.Span,
    spanName string,
) error {
```

**Comment by aryszka** [comment]
:+1:

File: filters/auth/authclient.go [hunk 4]
```diff
@@ -129,11 +137,24 @@ func jsonGet(url *url.URL, accessToken string, doc interface{}, client *http.Cli
...
 	body := url.Values{}
 	body.Add(tokenKey, auth)
-
+	req, err := http.NewRequest("POST", u.String(), strings.NewReader(body.Encode()))
+	span := injectSpan(tracer, parentSpan, spanName, req)
+	if span != nil {
+		defer span.Finish()
+	}
 	rsp, err := client.PostForm(u.String(), body)
```
[the remark this answers is not in the document]
Comment (arjunrn) [reply filters/auth/authclient.go]: It should have been used. I've fixed it now.

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by arjunrn** [comment]
:+1:

**Pull request merged**